staging/src/k8s.io/apimachinery/pkg/util/managedfields/gvkparser.go

```diff
@@ -79,10 +80,29 @@ func NewGVKParser(models proto.Models, preserveUnknownFields bool) (*GvkParser,
 	return &parser, nil
 }
 
-// Get and parse GroupVersionKind from the extension. Returns empty if it doesn't have one.
-func parseGroupVersionKind(s proto.Schema) []schema.GroupVersionKind {
-	extensions := s.GetExtensions()
+func NewGVKParserFromOpenAPI(schemas map[string]*spec.Schema, preserveUnknownFields bool) (*GvkParser, error) {
```

Comment:
Hmm, I'm not a big fan of this being public and called only by the type converter. Eventually we're going to move the typeconverter and, because I've tried to do it already, it'll create a bad dependency cycle with this. It would be simpler to just copy the whole gvkparser into a new one that is private next to the typeconverter.

Reply:
refactored to do as you said, and remove usage of gvkparser by replacing it with a TypeResolver interface. should we also remove the gvkparser entirely?